compiler/visq/visqlib/QErrorComputer.py

```diff
@@ -198,3 +199,46 @@ def get_final_result(self):
     def run(self):
         self.advance_on(self._fp32_dir, self._fq_dir)
         return self.get_final_result()
+
+
+# Scaled Root Mean Square Error (SRMSE)
+# SRMSE = sqrt(MSE) / scale
+class SRMSEComputer(QErrorComputer):
+    def __init__(self, fp32_dir, fq_dir):
+        super().__init__(fp32_dir, fq_dir)
+        self.scale_file = Path(fq_dir) / 'scales.txt'
+
+    # Incrementally compute Qerror while traversing all data in fp32_dir and fq_dir
+    def advance_on(self, fp32_dir, fq_dir):
+        data_paths = self.collect_data_path(fp32_dir, fq_dir)
+
+        for tensor_name, data_path in data_paths.items():
+            for (fp32_data_path, fq_data_path) in data_path:
+                fp32_data = np.load(fp32_data_path)
+                fq_data = np.load(fq_data_path)
+
+                MSE = np.square(fp32_data - fq_data).mean()
+
+                if tensor_name in self.qerror_map:
+                    self.qerror_map[tensor_name] += MSE
+                else:
+                    self.qerror_map[tensor_name] = MSE
+
+    def get_final_result(self):
+        with open(self.scale_file) as f:
```

Comment:
do need to do file error check?

Reply:
I guess that file error means that file's existence or contents (including format).

1. Existence
- scale.txt is created by `DumpFakeQuantFM.py` which is always called before metric computation. So we can make sure that the file exists.

2. Contents
- scale.txt is created only by `DumpFakeQuantFM.py`, so the format is fixed (json).

Or do you mean other kinds of errors?

Reply:
Well... there are unexpected things that happen :)
We can go as-is and fix when bad things happen.